src/Library/Fsm/FSMBuilder.cpp

```diff
@@ -0,0 +1,66 @@
+#include "FSMBuilder.h"
+
+#include <Library/Logger/Logger.h>
+
+#include <memory>
+#include <utility>
+#include <string>
+
+FSMBuilder &FSMBuilder::state(std::string_view stateName, std::unique_ptr<FSMState> state) {
+    _latestOnTransition.clear();
+    auto entry = std::make_unique<FSM::StateEntry>();
+    entry->name = stateName;
+    entry->state = std::move(state);
+    _states.push_back(std::move(entry));
+    return *this;
+}
+
+FSMBuilder &FSMBuilder::on(std::string_view transitionName) {
+    _latestOnTransition.clear();
+    if (_states.empty()) {
+        logger->warning(FSM::fsmLogCategory, "Can't create a transition with name [{}]. No state has been setup in the FSMBuilder",
+            transitionName);
```

Comment:
I have a feeling these should be exceptions. Let's discuss.

Reply:
Yes, probably all the warnings logged in the Builder should throw or at least assert. They are mostly programmer errors so it makes sense.

Reply:
If we're exposing this to lua, then assertions are out of question though.

Reply:
I changed the log from warning to error ( because it is an error after all ) and added an assert. Maybe you meant real exceptions tho. Let me know.